modules/virtualization/microvm/common/dns-dhcp.nix

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+_: let
+  domain = "ghaf";
+in {
+  # Disable resolved since we are using dnsmasq
+  # This options defaults to false but has been
+  # tested not to work with dnsmasq if not set
+  # explicitly to false
+  services.resolved.enable = false;
+  # do not really use resolvconf for anything
+  networking.resolvconf.enable = false;
+
+  # dnsmasq prioritises /etc/hosts to which NixOS
+  # generates hostname entry. Here we generate
+  # additional hosts that is used to replace
+  # /etc/hosts
+  environment.etc."additional-hosts" = {
+    text = ''
+      127.0.0.1 localhost
+      192.168.100.1 net-vm
+    '';
+    mode = "0444";
+  };
+
+  # Dnsmasq is used as a DHCP/DNS server inside the NetVM
+  services.dnsmasq = {
+    enable = true;
+    settings = {
+      # keep local queries within domain by
+      # caching them within dnsmasq. query outside
+      # only if name is not available locally
+      server = ["192.168.100.1" "8.8.8.8"];
```

Comment:
Here we should list only upstream servers or specify that 192.168.100.1 is local server. E.g. like this
        server = ["8.8.8.8"];
        local = ["/${domain}/192.168.100.1"];
This resolves the CPU load problem. The local -part might be unnecessary.

Reply:
@riskuuse @vadika @leivos-unikie - addressed in [10c35fd](https://github.com/tiiuae/ghaf/commit/10c35fd5c342032ae4649c207e2e97a58d101b14)
Tested with "Testing"-method and no high CPU load observed with network disconnected.